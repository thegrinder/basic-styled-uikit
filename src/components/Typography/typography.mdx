---
name: Typography
menu: Components
route: /components/typography
---

import { Playground, Props } from 'docz';
import { BoxProvider, Box } from '../Box';
import TypographyProvider from './TypographyProvider';
import Text from './Text/Text';
import Heading from './Heading/Heading';


## **TypographyProvider**

`<TypographyProvider />` is responsible for passing down the theme to the `<Heading />` and `<Text />` component. It expects the theme to be of certain structure.


---


## **Heading**

Heading component grabs the theme given to `<TypographyProvider />` and applies the style defined there based on the given `color`, `sizing`, and `emphasis` prop values. It also accepts a few different non-theme related props for convenience reasons.

<Props of={Heading} />


---


## **Text**

Text component grabs the theme given to `<TypographyProvider />` and applies the style defined there based on the given `color`, `sizing`, and `emphasis` prop values. It also accepts a few different non-theme related props for convenience reasons.

<Props of={Text} />


---


### Example


<Playground>
  {() => {
    const boxTheme = {
      bgColors: {
        neutral: '#fff',
        primary: '#1e88e5',
        success: '#43a047',
        warning: '#fdd835',
        danger: '#f4511e',
      },
    };

    const typographyTheme = {
      baseFontSize: 16,
      baseLineHeight: 1.5,
      opacities: {
        high: 1,
        medium: 0.8,
        low: 0.6,
      },
      colors: {
        neutral: '#161616',
        primary: '#1e88e5',
        success: '#43a047',
        warning: '#fdd835',
        danger: '#f4511e',
      },
      onBackground: {
        primary: {
          neutral: '#fff',
        },
        success: {
          neutral: '#fff',
        },
        warning: {
          neutral: '#161616',
        },
        danger: {
          neutral: '#fff',
        },
      },
      text: {
        sizings: {
          l: {
            fontSize: '1.25rem',
          },
          m: {
            fontSize: '1rem',
          },
          s: {
            fontSize: '.875rem',
          },
          xs: {
            fontSize: '.75rem',
          },
        },
        common: {
          fontFamily: `'Inter var', sans-serif`,
        },
      },
      heading: {
        sizings: {
          h1: {
            fontSize: '3rem',
          },
          h2: {
            fontSize: '2.5rem',
          },
          h3: {
            fontSize: '2rem',
          },
          h4: {
            fontSize: '1.5rem',
          },
          h5: {
            fontSize: '1.25rem',
          },
          h6: {
            fontSize: '1rem',
          },
        },
        common: {
          fontFamily: `'Inter var', sans-serif`,
          fontWeight: 600,
        },
      },
    };

    const Example = () => (
      <BoxProvider theme={boxTheme}>
        <TypographyProvider theme={typographyTheme}>
          <>
            <h2>Heading sizes:</h2>
            <Heading as="h1">h1 heading</Heading>
            <Heading as="h2">h2 heading</Heading>
            <Heading as="h3">h3 heading</Heading>
            <Heading as="h4">h4 heading</Heading>
            <Heading as="h5">h5 heading</Heading>
            <Heading as="h6">h6 heading</Heading>
            <Heading sizing="h6" as="h1" marginBottom>
              small h1 heading
            </Heading>

            <h2>Heading colors:</h2>
            <Heading as="h6" color="neutral">neutral heading</Heading>
            <Heading as="h6" color="primary">primary heading</Heading>
            <Heading as="h6" color="danger">danger heading</Heading>
            <Heading as="h6" color="warning">warning heading</Heading>
            <Heading as="h6" color="success" marginBottom>
              success heading
            </Heading>

            <h2>Heading colors on different backgrounds:</h2>
            <Box bgColor="neutral">
              <Heading as="h6">heading on neutral backgroud</Heading>
            </Box>
            <Box bgColor="primary">
              <Heading as="h6">heading on primary backgroud</Heading>
            </Box>
            <Box bgColor="success">
              <Heading as="h6">heading on success backgroud</Heading>
            </Box>
            <Box bgColor="warning">
              <Heading as="h6">heading on warning backgroud</Heading>
            </Box>
            <Box bgColor="danger" className="margin-bottom">
              <Heading as="h6">heading on danger backgroud</Heading>
            </Box>

            <h2>Text sizes:</h2>
            <Text sizing="l">large text</Text>
            <Text sizing="m">medium text</Text>
            <Text sizing="s">small text</Text>
            <Text sizing="xs" marginBottom>
              extra small text
            </Text>

            <h2>Text colors:</h2>
            <Text color="neutral">neutral text</Text>
            <Text color="primary">primary text</Text>
            <Text color="danger">danger text</Text>
            <Text color="warning">warning text</Text>
            <Text color="success">success text</Text>

            <h2>Text colors on different backgrounds:</h2>
            <Box bgColor="neutral">
              <Text>text on neutral backgroud</Text>
            </Box>
            <Box bgColor="primary">
              <Text>text on primary backgroud</Text>
            </Box>
            <Box bgColor="success">
              <Text>text on success backgroud</Text>
            </Box>
            <Box bgColor="warning">
              <Text>text on warning backgroud</Text>
            </Box>
            <Box bgColor="danger">
              <Text>text on danger backgroud</Text>
            </Box>

          </>
        </TypographyProvider>
      </BoxProvider>
    );

    return <Example />;
  }}
</Playground>
