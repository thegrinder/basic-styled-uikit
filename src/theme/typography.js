export const getBaseFontSize = ({ theme }) => theme.uiKit.typography.baseFontSize;

export const getBaseLineHeight = ({ theme }) => theme.uiKit.typography.baseLineHeight;

export const calcLineHeight = () => {};

const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
