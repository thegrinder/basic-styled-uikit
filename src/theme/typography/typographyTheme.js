const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
