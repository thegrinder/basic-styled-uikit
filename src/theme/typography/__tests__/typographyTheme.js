import typography from '../typographyTheme';

describe('typographyTheme', () => {
  it('should be defined', () => {
    expect(typography).toBeDefined();
    expect(typography).toMatchSnapshot();
  });
});
