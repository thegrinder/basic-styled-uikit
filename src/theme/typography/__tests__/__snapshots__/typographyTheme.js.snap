// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typographyTheme should be defined 1`] = `
Object {
  "baseFontSize": 16,
  "baseLineHeight": 1.5,
}
`;
